src/index.ts

```diff
@@ -63,6 +69,62 @@ const parseRecordValues = <K extends string>(
   )
 }
 
+const parseRecordValuesForHeaders = <U extends string>(
+  headersSchemaRecord: Record<U, D.Decoder<unknown, string>>,
+  headersRuntime: {
+    [key: string]: string | undefined
+  },
+  logStore: LogStore
+): taskEither.TaskEither<string, Record<U, string>> => {
+  return pipe(
+    headersSchemaRecord,
+    parTraverse((key, codec) => {
+      const decoderRecord = D.struct({ [key]: codec })
+      logStore.appendLog(['parsing headers: ', headersSchemaRecord])
+
+      return pipe(
+        parse(decoderRecord, { [key]: headersRuntime[key] }),
+        taskEither.fromEither,
+        taskEither.map((v) => {
+          logStore.appendLog(['parsed headers successfully!'])
+          return v[key]
+        })
+      )
+    }),
+    taskEither.mapLeft((e) => {
+      return `incorrect Headers runtime: ${draw(e)}`
+    })
+  )
+}
+
+const parseRecordValuesForQueryparams = <H extends string>(
```

Comment:
the way we decode a SchemaRecord is actually always the same, could we abstract it away in order to avoid this proliferation of duplicate logic?

Reply:
sure! Thank you 🙏🏼